### PR TITLE
test: E2E for muliple cluster resource snapshots for same group index

### PR DESCRIPTION
**Pull request by Arvindthiru**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: test/e2e/placement_selecting_resources_test.go [hunk 1]
```diff
@@ -1227,3 +1227,78 @@ var _ = Describe("validating CRP revision history allowing multiple revisions wh
...
+		// Delete the CRP.
+		crp := &placementv1beta1.ClusterResourcePlacement{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpName,
+			},
+		}
+		Expect(hubClient.Delete(ctx, crp)).To(Succeed(), "Failed to delete CRP %s", crpName)
+	})
+
+	It("should remove placed resources from all member clusters", checkIfRemovedWorkResourcesFromTwoMemberClusters)
```
Comment (zhiying-lin) [inline test/e2e/placement_selecting_resources_test.go:1298]: check if the secrets are removed?

File: pkg/controllers/clusterresourceplacement/controller.go [hunk 2]
```diff
@@ -35,8 +35,8 @@ import (
 )
 
 // The max size of an object in k8s is 1.5MB because of ETCD limit https://etcd.io/docs/v3.3/dev-guide/limit/.
-// We choose 1MB as the max size for all the selected resources within one clusterResourceSnapshot object.
-var resourceSnapshotResourceSizeLimit = 1024 * (1 << 10) // 1 MB
+// We choose 800 bytes as the max size for all the selected resources within one clusterResourceSnapshot object.
```
[the remark this answers is not in the document]
Comment (ryanzhang-oss) [reply pkg/controllers/clusterresourceplacement/controller.go:38]: it was 1MB which is about 1,000,000. How come we suddenly reduce this limit dramatically?

**Review by zhiying-lin — APPROVED** [review]

File: test/e2e/placement_selecting_resources_test.go [hunk 3]
```diff
@@ -1227,3 +1236,234 @@ var _ = Describe("validating CRP revision history allowing multiple revisions wh
...
+	for i := 0; i < 3; i++ {
+		var secret corev1.Secret
+		Expect(utils.GetObjectFromManifest("../integration/manifests/resources/test-large-secret.yaml", &secret)).Should(Succeed(), "Failed to read large secret from file")
+		secret.Namespace = workNamespace().Name
+		secret.Name = fmt.Sprintf(appSecretNameTemplate, i)
+		Expect(hubClient.Create(ctx, &secret)).To(Succeed(), "Failed to create secret %s/%s", secret.Name, secret.Namespace)
+	}
+
+	// sleep for 5 seconds to ensure secrets exist to prevent flake.
+	<-time.After(5 * time.Second)
```
Comment (ryanzhang-oss) [inline test/e2e/placement_selecting_resources_test.go:1330]: using time is a sure way to introduce flakiness in test

Comment (ryanzhang-oss) [reply test/e2e/placement_selecting_resources_test.go:1330]: why not add an eventual loop to make sure we can get the secrets?

Comment (zhiying-lin) [reply test/e2e/placement_selecting_resources_test.go:1330]: The comment is not 100% correct, the sleep is introduced to make sure the secrets are detected by our resource detector controller. Arvind found that our controller sometimes detected the first two secrets first and then the other one.

**Pull request merged**